### PR TITLE
Add repository validation script MVP implementation

### PR DESCRIPTION
**Pull request by LarsArtmann**
Original issue: https://github.com/LarsArtmann/mono/issues/57

### CODE REVIEW

File: repo-validation/internal/config/config.go [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+// GetDockerFiles returns the list of Docker-related files
+func GetDockerFiles() []FileRequirement {
+	return []FileRequirement{
+		{
+			Path:        "Dockerfile",
+			Category:    "Docker",
+			Priority:    "Must-have",
+			Description: "Instructions for building a Docker image for the application",
+			TemplatePath: "", // No template for MVP
+		},
```
[the remark this answers is not in the document]
Comment (LarsArtmann) [reply repo-validation/internal/config/config.go:155]: @coderabbitai link this to https://github.com/LarsArtmann/mono/issues/66

File: repo-validation/internal/checker/checker.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
+package checker
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/LarsArtmann/templates/repo-validation/internal/config"
+)
```
[the remark this answers is not in the document]
Comment (LarsArtmann) [reply repo-validation/internal/checker/checker.go:11]: NO!

**Comment by LarsArtmann** [comment]
@coderabbitai configuration

**Comment by LarsArtmann** [comment]
@coderabbitai review

**Comment by LarsArtmann** [comment]
I've created issue #17 to track the need for splitting the LICENSE.md.tmpl file into three separate templates (Proprietary, MIT, and EUPL) to provide more flexibility for different project types.

File: repo-validation/main.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+		if *jsonOutput {
+			// Output error in JSON format
+			fmt.Printf("{\"error\": \"%s\", \"code\": %d}\n", err.Error(), exitCode)
+		} else {
+			// Output error in human-readable format
+			fmt.Fprintf(os.Stderr, "Error: %v (code: %d)\n", err, exitCode)
+		}
+		os.Exit(exitCode)
+	}
```
[the remark this answers is not in the document]
Comment (LarsArtmann) [reply repo-validation/main.go:97]: Use the log lib

[the remark this answers is not in the document]
Comment (LarsArtmann) [reply repo-validation/main.go:97]: "github.com/charmbracelet/log"!!!!

**Comment by LarsArtmann** [comment]
# PR Improvements Summary

I've addressed all the feedback from the PR review and made several significant improvements to the repo-validation script. Here's a summary of the changes:

## Code Structure and Organization

1. **Functional Options Pattern**
   - Refactored the configuration handling to use the functional options pattern
   - Added extension points for future configuration options
   - Improved code readability and maintainability

2. **Modular Code Structure**
   - Split main.go into multiple files for better organization
   - Created separate cmd package with run.go, prompt.go, and validate.go
   - Improved separation of concerns

## Error Handling and Reporting

1. **Comprehensive Error Collection**
   - Now collecting all errors instead of stopping at the first one
   - Providing users with a complete picture of validation issues

2. **Specific Exit Codes**
   - Added detailed exit codes for different error types:
     - 0: Success
     - 1: General error
     - 2: Path resolution error
     - 3: Missing must-have files
     - 4: Invalid configuration options
     - 5: File access or permission error

3. **Colorized Error Output**
   - Added colorized error output using the Charmbracelet log package
   - Improved readability of error messages

## Interactive Mode Improvements

1. **Better User Experience**
   - Added checks to prevent prompting in JSON output mode
   - Added checks for non-interactive terminals
   - Improved conflict resolution for incompatible flags

## Documentation

1. **Enhanced README**
   - Added quick install one-liner at the top
   - Added common workflow examples with flag combinations
   - Updated exit code documentation

## Testing

1. **Comprehensive Test Coverage**
   - Added unit tests for all packages:
     - checker
     - config
     - errors
     - exitcode
     - reporter
   - Testing edge cases and error conditions

## Consistency

1. **Consistent Code Style**
   - Replaced hardcoded strings with constants
   - Standardized error handling across the codebase
   - Improved JSON output handling

All review comments have been addressed, and the code is now more robust, maintainable, and user-friendly.
